buildSrc/src/main/java/org/opensearch/gradle/testclusters/OpenSearchCluster.java

```diff
@@ -124,12 +134,32 @@ public void setNumberOfNodes(int numberOfNodes) {
             );
         }
 
+        if (numberOfNodes < zoneCount) {
+            throw new IllegalArgumentException(
+                "Number of nodes should be >= zoneCount but was " + numberOfNodes + " for " + this.zoneCount
+            );
+        }
+
+        int currentZone = 1;
         for (int i = nodes.size(); i < numberOfNodes; i++) {
```

Comment:
I believe the code will be much cleaner (and less cryptic) if we clearly branch off "with zone" / "without zone" flows

```
if (hasZoneProperty()) {
  int currentZone = 1;
 for (int i = nodes.size(); i < numberOfNodes; i++) {
 ...
 }
} else {
  for (int i = nodes.size(); i < numberOfNodes; i++) {
          addNode(clusterName + "-" + i, "" );
  }
}
```

Reply:
Done.

Reply:
TY